src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/security/AuthorisedCaseSearchOperation.java

```diff
@@ -83,35 +95,75 @@ private CaseSearchResult searchCasesAndFilterFieldsByAccess(List<CaseType> autho
 
         return result;
     }
 
-    private void filterFieldsByAccess(List<CaseType> authorisedCaseTypes, List<CaseDetails> cases) {
+    private void filterFieldsByAccess(List<CaseType> authorisedCaseTypes, List<CaseDetails> cases, CrossCaseTypeSearchRequest authorisedSearchRequest) {
         Map<String, CaseType> caseTypeIdByCaseType = authorisedCaseTypes
             .stream()
             .collect(Collectors.toMap(CaseType::getId, Function.identity()));
 
         cases.stream()
             .filter(caseDetails -> caseTypeIdByCaseType.containsKey(caseDetails.getCaseTypeId()))
-            .forEach(caseDetails -> {
-                filterCaseFieldsByAclAccess(caseTypeIdByCaseType.get(caseDetails.getCaseTypeId()), caseDetails);
-                filterCaseFieldsBySecurityClassification(caseDetails);
-            });
+            .forEach(caseDetails -> filterCaseData(caseTypeIdByCaseType.get(caseDetails.getCaseTypeId()), caseDetails, authorisedSearchRequest));
+    }
+
+    private void filterCaseData(CaseType authorisedCaseType, CaseDetails caseDetails, CrossCaseTypeSearchRequest authorisedSearchRequest) {
+        filterCaseDataByAclAccess(authorisedCaseType, caseDetails);
+        filterCaseFieldsBySecurityClassification(caseDetails);
+        transformCaseDataForMultiCaseTypeSearch(authorisedSearchRequest, authorisedCaseType, caseDetails);
+    }
+
+    private void filterCaseDataByAclAccess(CaseType authorisedCaseType, CaseDetails caseDetails) {
+        JsonNode caseData = convertCaseDataToJsonNode(caseDetails);
+        JsonNode accessFilteredData = accessControlService.filterCaseFieldsByAccess(caseData, authorisedCaseType.getCaseFields(), getUserRoles(), CAN_READ);
+        caseDetails.setData(convertJsonNodeToCaseData(accessFilteredData));
     }
 
     private void filterCaseFieldsBySecurityClassification(CaseDetails caseDetails) {
         classificationService.applyClassification(caseDetails);
     }
 
-    private void filterCaseFieldsByAclAccess(CaseType authorisedCaseType, CaseDetails caseDetails) {
-        JsonNode data = objectMapperService.convertObjectToJsonNode(caseDetails.getData());
-        JsonNode filteredData = accessControlService.filterCaseFieldsByAccess(data, authorisedCaseType.getCaseFields(), getUserRoles(), CAN_READ);
-        caseDetails.setData(objectMapperService.convertJsonNodeToMap(filteredData));
+    private void transformCaseDataForMultiCaseTypeSearch(CrossCaseTypeSearchRequest searchRequest, CaseType authorisedCaseType, CaseDetails caseDetails) {
+        if (searchRequest.isMultiCaseTypeSearch() && caseDetails.getData() != null) {
+            JsonNode caseData = convertCaseDataToJsonNode(caseDetails);
+            List<String> aliasFields = searchRequest.getSourceFilterAliasFields();
+            JsonNode filteredMultiCaseTypeSearchData = objectMapperService.createEmptyJsonNode();
+
+            authorisedCaseType.getSearchAliasFields()
```

Comment:
very long block of fuctional operations. Easy to get lost. Can we split is separate smaller pieces with self explanatory names please

Reply:
simplified